src/core/lib/surface/channel.cc

```diff
@@ -122,11 +122,15 @@ grpc_channel* grpc_channel_create_with_builder(
           static_cast<uint32_t>(args->args[i].value.integer) |
           0x1; /* always support no compression */
     } else if (0 == strcmp(args->args[i].key, GRPC_ARG_CHANNELZ_CHANNEL_NODE)) {
-      GPR_ASSERT(args->args[i].type == GRPC_ARG_POINTER);
```

Comment:
@kcwu Could you point to me the evidence that proves this assert is the rootcause? I looked at the log but the stacktrace terminates at https://chromium.googlesource.com/external/github.com/grpc/grpc/+/74b981a6a3d9ba17f3acae1d72b9109325ef656d/src/core/lib/surface/channel.cc#260

Reply:
I guess it was optimized, so you cannot see the last frame from the log.
I build the fuzzer and actually run gdb with the input to find it. If you want to see it, one evidence is open the minimized input file and you can see the key string, which is matching line 124.

```
$ hexdump -C clusterfuzz-testcase-minimized-grpc_api_fuzzer-5664490786258944
(skip some lines)
00000220  01 01 01 01 67 72 70 63  2e 63 68 61 6e 6e 65 6c  |....grpc.channel|
00000230  7a 5f 63 68 61 6e 6e 65  6c 5f 6e 6f 64 65 01 01  |z_channel_node..|
```

Reply:
Here are my gdb output
```
$ gdb -q --args ./out/libfuzzer-ar/grpc_api_fuzzer clusterfuzz-testcase-minimized-grpc_api_fuzzer-5664490786258944
Reading symbols from ./out/libfuzzer-ar/grpc_api_fuzzer...done.
(gdb) r
Starting program: /btrfs/chromium/fuzzing/src/out/libfuzzer-ar/grpc_api_fuzzer clusterfuzz-testcase-minimized-grpc_api_fuzzer-5664490786258944
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
INFO: Seed: 748101600
INFO: Loaded 1 modules   (50669 inline 8-bit counters): 50669 [0x5555561223f0, 0x55555612e9dd),
INFO: Loaded 1 PC tables (50669 PCs): 50669 [0x55555612e9e0,0x5555561f48b0),
/btrfs/chromium/fuzzing/src/out/libfuzzer-ar/grpc_api_fuzzer: Running 1 inputs 1 time(s) each.
Running: clusterfuzz-testcase-minimized-grpc_api_fuzzer-5664490786258944

Thread 1 "grpc_api_fuzzer" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50      ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff708d8ad in __GI_abort () at abort.c:79
#2  0x0000555555d9861c in grpc_channel_create_with_builder (builder=<optimized out>, channel_stack_type=<optimized out>) at ../../third_party/grpc/src/src/core/lib/surface/channel.cc:126
#3  0x0000555555d98cee in grpc_channel_create (target=0x6020000040f0 "dns:grpc.qqbaad", input_args=<optimized out>, channel_stack_type=GRPC_CLIENT_CHANNEL, optional_transport=0x0, resource_user=<optimized out>) at ../../third_party/grpc/src/src/core/lib/surface/channel.cc:260
#4  0x0000555555c9fa66 in grpc_core::Chttp2InsecureClientChannelFactory::CreateChannel (this=<optimized out>, target=<optimized out>, args=<optimized out>) at ../../third_party/grpc/src/src/core/ext/transport/chttp2/client/insecure/channel_create.cc:65
#5  0x0000555555c9f573 in grpc_insecure_channel_create (target=<optimized out>, args=<optimized out>, reserved=<optimized out>) at ../../third_party/grpc/src/src/core/ext/transport/chttp2/client/insecure/channel_create.cc:101
#6  0x0000555555ab583a in LLVMFuzzerTestOneInput (data=<optimized out>, size=<optimized out>) at ../../third_party/grpc/src/test/core/end2end/fuzzers/api_fuzzer.cc:801
#7  0x0000555555b150c0 in fuzzer::Fuzzer::ExecuteCallback (this=0x617000000080, Data=<optimized out>, Size=<optimized out>) at ../../third_party/libFuzzer/src/FuzzerLoop.cpp:553
#8  0x0000555555ae2aeb in fuzzer::RunOneTest (F=<optimized out>, InputFilePath=<optimized out>, MaxLen=<optimized out>) at ../../third_party/libFuzzer/src/FuzzerDriver.cpp:292
#9  0x0000555555ae858b in fuzzer::FuzzerDriver (argc=<optimized out>, argv=<optimized out>, Callback=<optimized out>) at ../../third_party/libFuzzer/src/FuzzerDriver.cpp:775
#10 0x0000555555b29ebd in main (argc=2, argv=0x7fffffffcd38) at ../../third_party/libFuzzer/src/FuzzerMain.cpp:19
(gdb) frame 2
#2  0x0000555555d9861c in grpc_channel_create_with_builder (builder=<optimized out>, channel_stack_type=<optimized out>) at ../../third_party/grpc/src/src/core/lib/surface/channel.cc:126
126           GPR_ASSERT(args->args[i].value.pointer.p != nullptr);
```